xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-api/src/main/java/org/xwiki/mentions/script/MentionsScriptService.java

```diff
@@ -98,13 +98,29 @@ public boolean isQuoteActivated()
     /**
      * Format a user mention.
      *
-     * @see MentionsFormatter#formatMention(String, DisplayStyle)
      * @param userReference the user reference
      * @param style the display style
      * @return the formatted mention
+     * @see MentionsFormatter#formatMention(String, DisplayStyle)
      */
     public String format(String userReference, DisplayStyle style)
     {
-        return this.formatter.formatMention(userReference, style);
+        return format(userReference, style, null);
```

Comment:
So that's actually where there's still a risk to call the FormatterProvider with a null type: shouldn't we deprecate this method, and force it to "user" instead of null? It should be enough no, in that case we shouldn't even need to handle the special case of a null type in the FormatterProvider, no?

Reply:
I'm ok with deprecating this format method, but I still think the special case should be handled in `FormatterProvider`, because it is not only used by the script service, but also by the renderer used to generate the mentions quotes.
We could also move the substitution of `null` to `"user"` outside of `FormatterProvider`, but that would lead to a small code duplication.

Reply:
> but also by the renderer used to generate the mentions quotes

IMO there should be a check here too to use "user" in case null is encountered. Basically as much as possible I'd avoid propagating a null value. I'm ok to keep checking it on the FormatterProvider for bulletproof reasons though.